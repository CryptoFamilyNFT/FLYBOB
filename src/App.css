.App-blurred {
  position: relative;
  text-align: center;
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-flow: column;
  overflow: hidden;
}

/* Stile dello sfondo sfocato */
.App-blurred::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-image: url("./assets/images/logo.jpeg");
  filter: blur(8px); /* Puoi regolare l'intensità della sfocatura qui */
  z-index: -1; /* Posiziona lo sfondo sfocato dietro il contenuto */
}

/* Stile per il contenuto dell'app */
.App-content {
  z-index: 1; /* Posiziona il contenuto sopra lo sfondo sfocato */
  backdrop-filter: blur(8px); /* Puoi regolare l'intensità della sfocatura qui */
  background-color: rgba(255, 255, 255, 0.5); /* Fai in modo che l'effetto di sfocatura sia più evidente */
}

@media screen and (min-width: 768px) {
  .App-blurred {
    flex-flow: row;
  }
}