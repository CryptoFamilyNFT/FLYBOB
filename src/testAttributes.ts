const TEST_ATTRIBUTES = {
  APP_COMPONENT: 'App',
  POSTER_COMPONENT: 'Poster',
  MENU_COMPONENT: 'Menu',
  BACK_BUTTON: 'Menu__BackButton',
  LICENSE_LINK: 'Menu__LicenseLink',
  REPO_LINK: 'Menu__RepoLink'
};

export default TEST_ATTRIBUTES;
