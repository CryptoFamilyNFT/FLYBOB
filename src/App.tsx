// Libs
import React from 'react';
import {BrowserRouter} from 'react-router-dom';

// Styles
import './App.css';

// Components
import Menu from './containers/Menu';
import Poster from './components/Poster';

// Constants
import testAttributes from "./testAttributes";

function App() {
  return (
    <BrowserRouter>
      <div className="App" data-testid={testAttributes.APP_COMPONENT}>
        <Poster/>
        <Menu/>
      </div>
    </BrowserRouter>
  );
}

export default App;
