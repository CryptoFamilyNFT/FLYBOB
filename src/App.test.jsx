import React from "react";

// Libs
import renderer from "react-test-renderer";
import {render, screen} from "@testing-library/react";
import '@testing-library/jest-dom';

// Components
import App from "./App";

// Constants
import testAttributes from "./testAttributes";

describe('<App />', () => {
  beforeEach(() => {
    render(<App />);
  })

  it('should renders correctly', () => {
    const tree = renderer
      .create(<App/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders component',  () => {
    expect(
      screen.getByTestId(testAttributes.APP_COMPONENT)
    ).toBeInTheDocument();
  });

  it('should contains <Poster /> component', () => {
    expect(
      screen.getByTestId(testAttributes.POSTER_COMPONENT)
    ).toBeInTheDocument();
  });

  it('should contains <Menu /> component', () => {
    expect(
      screen.getByTestId(testAttributes.MENU_COMPONENT)
    )
  });
});

