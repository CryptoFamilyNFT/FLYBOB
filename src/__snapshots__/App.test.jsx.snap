// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should renders correctly 1`] = `
<div
  className="App"
  data-testid="App"
>
  <div
    className="Poster"
    data-testid="Poster"
  >
    <img
      alt="Logo"
      loading="lazy"
      src="logo.png"
    />
  </div>
  <div
    className="Menu"
    data-testid="Menu"
  >
    <button
      data-testid="Menu__BackButton"
      hidden={true}
      onClick={[Function]}
    >
      Back
    </button>
    <div
      className="List"
    >
      <button>
        <a
          aria-current={null}
          href="/game"
          onClick={[Function]}
        >
          Play
        </a>
      </button>
      <button>
        <a
          aria-current={null}
          href="/author"
          onClick={[Function]}
        >
          Author
        </a>
      </button>
    </div>
    <div
      className="Links"
    >
      <a
        className="License"
        data-testid="Menu__LicenseLink"
        href="https://github.com/corocoto/flappy-bird/blob/master/LICENSE"
        rel="noopener noreferrer license"
        target="_blank"
        title="Link on the licence"
      >
         
      </a>
      <a
        className="Repo"
        data-testid="Menu__RepoLink"
        href="https://github.com/corocoto/flappy-bird"
        rel="noopener noreferrer bookmark"
        target="_blank"
        title="Link on the repository"
      >
         
      </a>
    </div>
  </div>
</div>
`;
